### DISCUSSION TITLE
Hot scale not working in VMware because Cloudstack not update correct Guest OS

### DISCUSSION THREAD

**Discussion by leduyquy** [apache/cloudstack#9861] [Q&A] [answered]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Instance
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.1
~~~

##### OS / ENVIRONMENT

Ubuntu 22.04 for Cloudstack management
vSphere 8.0
Ubuntu 22.04 for Instance

##### SUMMARY

In template and Instance I set correct OS type, But when Cloudstack clone VM from template it not set correct to Guest OS Family and Guest OS Version on VM Options of vm.

 It set to default:
Guest OS Family: Other
Guest OS Version: Other (64-bit)

With default options, It only allow scale CPU, I can't hot scale RAM with default options. I can try edit manual Guest OS Family and version to correct os type at vmware to be able to hot scale CPU and VMware but when i change computer offering, Cloudstack overwrite it to default. 

I can try import VM and enable read config from OVA to set correct OS info when clone VM. However, with the above method  i can set root disk for vm when deploy. It not friendly for client.

##### STEPS TO REPRODUCE

~~~
Create Template and set correct OS for template 
Clone VM from template
~~~

##### EXPECTED RESULTS

~~~
Guest OS Famlly and Version in  VM Options of VM update correct OS and can hot scale CPU and RAM.
~~~

##### ACTUAL RESULTS

~~~
Guest OS Famlly and Version set to default value and can't hot scale CPU and RAM.
~~~

**Comment by weizhouapache** [comment]
@leduyquy 

what's your vmware version ?

can you share the correct os type you want to use ?

**Comment by leduyquy** [comment]
Hi @weizhouapache 
I use vSphere 8.0.0 and VMware esxi 8.0.2

Example options I set to VM on VMware:

Ubuntu:
Guest OS Family: Linux
Guest OS Version: Ubuntu Linux (64bit)

Windows:
Guest OS Family: Windows
Guest OS Version: version OS template, eg: Microsoft Windows Server 2022 (64-bit)

**Comment by weizhouapache** [comment, marked as answer]
@leduyquy 
they are not officially supported in 4.19.x

can add the SQL changes manually and retry ?
 https://github.com/apache/cloudstack/pull/9591
